### PR TITLE
Show nomination balance below nominees

### PR DESCRIPTION
**Pull request by Tbaut**
closes https://github.com/polkadot-js/apps/issues/2085

Not sure if there is any more straightforward way, this is very convoluted and seems costly.
I couldn't test with more than 1 active nominees yet, I'll get there once my node is done compiling.

### CODE REVIEW

File: packages/page-staking/src/Actions/Account/ListNominees.tsx [hunk 1]
```diff
@@ -18,6 +20,39 @@ function ListNominees ({ nominating, stashId }: Props): React.ReactElement<Props
...
   const { api } = useApi();
   const { nomsActive, nomsChilled, nomsInactive, nomsOver, nomsWaiting } = useInactives(stashId, nominating);
+  const sessionInfo = useCall<DeriveSessionProgress>(api.query.staking && api.derive.session?.progress);
+  const [nomBalanceMap, setNomBalanceMap] = useState<Record<string, BN>>({});
+
+  useEffect(() => {
+    let unsub: (() => void) | undefined;
+
+    if (nomsActive && sessionInfo?.activeEra) {
+      api.derive.staking.eraExposure(sessionInfo.activeEra, (v) => {
```
Comment (jacogr) [inline packages/page-staking/src/Actions/Account/ListNominees.tsx:30]: Would prefer just using a `useCall` with a transform. (Since we don't need to manage state, as it stands this also doesn't take care of unmounts, which useCall does without worries)

**Comment by jacogr** [comment]
Will check on mine, have some with multiple nominations.

**Comment by jacogr** [comment]
Ok, it does show multiples, but it flickers on block updates and seems to change position. (no index)

File: packages/page-staking/src/Actions/Account/ListNominees.tsx (hunk 1 above)
Comment (jacogr) [reply packages/page-staking/src/Actions/Account/ListNominees.tsx:30]: Probably not with a transfers - useCall to retrieve the data and then a useMemo to combine it. (Since otherwise the active noms will be captured in the closure, so the memo will take both the era and noms in to get a result)

But yes, quite heavy atm since it subscribes/unsubscribes all the time.

**Comment by Tbaut** [comment]
Thanks, will try to implement those inputs!

File: packages/page-staking/src/Actions/Account/ListNominees.tsx [hunk 2]
```diff
@@ -17,6 +19,29 @@ function ListNominees ({ nominating, stashId }: Props): React.ReactElement<Props
   const { t } = useTranslation();
   const { api } = useApi();
   const { nomsActive, nomsChilled, nomsInactive, nomsOver, nomsWaiting } = useInactives(stashId, nominating);
+  const sessionInfo = useCall<DeriveSessionProgress>(api.query.staking && api.derive.session?.progress);
+  const eraExposure = useCall<DeriveEraExposure>(api.derive.staking.eraExposure, [sessionInfo?.activeEra]);
```
Comment (jacogr) [inline packages/page-staking/src/Actions/Account/ListNominees.tsx:23]: This is one of those problematic ones that may break on Edgeware, it doesn't have that functionality. (Not sure if the derive itself does the check, it should...)

**Comment by Tbaut** [comment]
No more flickering afaics.
> seems to change position. (no index).

Not sure I get that, the nominator ordering should not be changed by this PR since it comes from the `nmsActive` array ([here](https://github.com/polkadot-js/apps/pull/3714/files#diff-020b090863fcb404663ca3419356a561R70)). Or did I misunderstand?

File: packages/page-staking/src/Actions/Account/ListNominees.tsx (hunk 2 above)
Comment (Tbaut) [reply packages/page-staking/src/Actions/Account/ListNominees.tsx:23]: Oh thanks, checking that (and will add it in the derive if it's not there)

**Comment by jacogr** [comment]
I think I meant to say "key or ordering". Lemme see the current version, couldn't quite put my finger on what was hapenning with the previous.

File: packages/page-staking/src/Actions/Account/ListNominees.tsx [hunk 3]
```diff
@@ -17,6 +19,29 @@ function ListNominees ({ nominating, stashId }: Props): React.ReactElement<Props
   const { t } = useTranslation();
   const { api } = useApi();
   const { nomsActive, nomsChilled, nomsInactive, nomsOver, nomsWaiting } = useInactives(stashId, nominating);
+  const sessionInfo = useCall<DeriveSessionProgress>(api.query.staking && api.derive.session?.progress);
+  const eraExposure = useCall<DeriveEraExposure>(api.derive.staking?.eraExposure, [sessionInfo?.activeEra]);
```
Comment (jacogr) [inline packages/page-staking/src/Actions/Account/ListNominees.tsx:23]: Cater for older chains.

```suggestion
  const eraExposure = useCall<DeriveEraExposure>(api.query.staking.erasStakers && api.derive.staking.eraExposure, [sessionInfo?.activeEra]);
```

File: packages/page-staking/src/Actions/Account/ListNominees.tsx [hunk 4]
```diff
@@ -44,9 +69,10 @@ function ListNominees ({ nominating, stashId }: Props): React.ReactElement<Props
         >
           {nomsActive.map((nomineeId, index): React.ReactNode => (
             <AddressMini
+              balance={nomBalanceMap[nomineeId]}
               key={index}
               value={nomineeId}
-              withBalance={false}
+              withBalance
```
Comment (jacogr) [inline packages/page-staking/src/Actions/Account/ListNominees.tsx:75]:
```suggestion
              withBalance={!!eraExposure}
```

File: packages/page-staking/src/Actions/Account/ListNominees.tsx (hunk 3 above)
Comment (Tbaut) [reply packages/page-staking/src/Actions/Account/ListNominees.tsx:23]: Thanks, I was digging through the api and was having trouble finding what it should be replaced by. Pretty hard to follow all these changes, derive are extremely valuable for that.

File: packages/page-staking/src/Actions/Account/ListNominees.tsx [hunk 5]
```diff
@@ -17,6 +19,29 @@ function ListNominees ({ nominating, stashId }: Props): React.ReactElement<Props
...
+  const eraExposure = useCall<DeriveEraExposure>(api.derive.staking?.eraExposure, [sessionInfo?.activeEra]);
+  const nomBalanceMap: Record<string, BN> = useMemo(() => {
+    const res: Record<string, BN> = {};
+
+    if (nomsActive && eraExposure) {
+      // for every active nominee
+      nomsActive.forEach((nom) => {
+        // cycle through its nominator to find our current stash
+        eraExposure.validators?.[nom].others.some((o) => {
+          if (o.who.toString() === stashId) {
```
Comment (jacogr) [inline packages/page-staking/src/Actions/Account/ListNominees.tsx:32]:
```suggestion
          if (o.who.eq(stashId)) {
```

**Review by jacogr — COMMENTED** [review]
Some small comments to cater for eg. Edgeware

File: packages/page-staking/src/Actions/Account/ListNominees.tsx (hunk 5 above)
Comment (Tbaut) [reply packages/page-staking/src/Actions/Account/ListNominees.tsx:32]: TIL again, didn't think that would work with BN and string!

Comment (jacogr) [reply packages/page-staking/src/Actions/Account/ListNominees.tsx]: who is AccountId, all the polkadot-js Codec stuff has .eq on them. (And in this case can compare hex, Uint8array, other AccountId and ss58)

**Review by jacogr — APPROVED** [review]
Thank you!

**Comment by Tbaut** [comment]
Thank YOU :hand:
